tests/phpunit/elementor/modules/admin-bar/test-module.php

```diff
@@ -0,0 +1,106 @@
+<?php
+namespace Elementor\Testing\Modules\AdminBar;
+
+use Elementor\Plugin;
+use Elementor\Modules\AdminBar\Module;
+use Elementor\Testing\Elementor_Test_Base;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Elementor_Test_Module extends Elementor_Test_Base {
+	/**
+	 * @var Module
+	 */
+	protected $module;
+
+	protected $fake_document_data = [
+		'meta_input' => [
+			'_elementor_edit_mode' => true,
+			'_elementor_data' => '[{"id":"378e77c", "elType": "section"}]',
+		],
+	];
+
+	public function setUp() {
+		parent::setUp();
+
+		wp_set_current_user( self::factory()->get_administrator_user()->ID );
+
+		$this->module = new Module();
+	}
+
+	public function test_it_should_returns_a_valid_config() {
+		$active_document = $this->create_document();
+		$document = $this->create_document();
+
+		query_posts( [ 'p' => $active_document->get_id() ] );
+
+		$frontend = Plugin::$instance->frontend;
+
+		$frontend->get_builder_content( $active_document->get_id() );
+		$frontend->get_builder_content( $document->get_id() );
+
+		$config = $this->module->get_admin_bar_config();
+
+		$this->assertTrue( isset( $config['elementor_app_site_editor'] ) );
+
+		$editButtonConfig = $config['elementor_app_site_editor'];
+
+		$this->assertEquals( $active_document->get_edit_url(), $editButtonConfig['href'] );
+
+		$this->assertCount( 2, $editButtonConfig['children'] );
+		$this->assertEquals( "elementor_edit_doc_{$document->get_id()}", $editButtonConfig['children'][ $document->get_id() ]['id'] );
+		$this->assertEquals( "elementor_app_site_editor", $editButtonConfig['children'][ intval( $document->get_id() ) + 1 ]['id'] );
+	}
+
+	public function test_it_should_not_add_document_that_is_not_valid() {
+		$excerpt_document = $this->create_document();
+		$not_supported_document = $this->create_document( [
+			'meta_input' => [
+				'_elementor_template_type' => 'not-supported'
+				// The prop 'show_on_admin_bar' of NotSupport document type is false.
+			],
+		] );
+
+		$frontend = Plugin::$instance->frontend;
+
+		$frontend->start_excerpt_flag( $excerpt_document->get_id() );
+		$frontend->get_builder_content( $excerpt_document->get_id() );
+		$frontend->end_excerpt_flag( $excerpt_document->get_id() );
+
+		$frontend->get_builder_content( $not_supported_document->get_id() );
+
+		$config = $this->module->get_admin_bar_config();
+
+		$editButtonConfig = $config['elementor_app_site_editor'];
+
+		$this->assertCount( 1, $editButtonConfig['children'] );
+		$this->assertEquals( "elementor_app_site_editor", $editButtonConfig['children'][0]['id'] );
+		$this->assertFalse( isset( $editButtonConfig['href'] ) );
```

Comment:
`assertNotContains`?
anyway, don't trust the `elementor_app_site_editor` item as an indicator (here and next text) . it may be removed.